staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -58,6 +58,29 @@ var (
 		},
 		[]string{"operation", "type"},
 	)
+
+	etcdWatcherChannelLength = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name: "etcd_watcher_channel_length",
+			Help: "Etcd watcher channel length",
+		},
+		[]string{"key", "channel"},
+	)
+	etcdEventsCounter = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name: "etcd_watcher_received_events",
+			Help: "Counter of events received from etcd broken by key",
+		},
+		[]string{"key"},
+	)
+	eventsSentLatency = compbasemetrics.NewHistogramVec(
+		&compbasemetrics.HistogramOpts{
+			Name:    "etcd_watcher_sended_events_latency_milliseconds",
+			Help:    "Latency bucket of watchers sended by key",
+			Buckets: compbasemetrics.DefBuckets,
+		},
```

Comment:
Do you mean:
etcd_watcher_sended_events_latency_milliseconds -> etcd_watcher_events_sent_latency_milliseconds?
Latency bucket of watchers sended by key -> Latency bucket of watchers sent event broken by key?

Reply:
Refined:
```go
etcdEventsSentLatency = compbasemetrics.NewHistogramVec(
		&compbasemetrics.HistogramOpts{
			Name:    "etcd_watcher_events_sent_latency_milliseconds",
			Help:    "Latency bucket of etcd watchers sent events broken by resource key",
			Buckets: compbasemetrics.DefBuckets,
		},
		[]string{"resource"},
	)
```